app/grandchallenge/reader_studies/views.py

```diff
@@ -1047,20 +1060,11 @@ def get_queryset(self):
                 answers__creator=self.request.user,
                 answer_count__gte=answerable_question_count,
             )
-        return queryset
+            if reader_study and reader_study.shuffle_hanging_list:
+                pks = queryset.values_list("pk", flat=True)
```

Comment:
Doesn't this fetch the queryset twice as it's done above in L1045?

Reply:
This is the filtered list, so it's a different list. In order to retain the proper index we need to filter the filtered values out of the initial full queryset. I hope that makes sense.

Reply:
Would be worth adding some comments to help us in the future.